src/query/expression/src/converts/arrow2/from.rs

```diff
@@ -781,6 +782,21 @@ impl Column {
                         unsafe { std::mem::transmute::<Buffer<i64>, Buffer<u64>>(offsets) };
                     Column::Geometry(BinaryColumn::new(arrow_col.values().clone(), offsets))
                 }
+                (DataType::Geography, ArrowDataType::LargeBinary) => {
+                    let arrow_col = arrow_col
+                        .as_any()
+                        .downcast_ref::<databend_common_arrow::arrow::array::BinaryArray<i64>>()
+                        .expect(
+                            "fail to read `Geography` from arrow: array should be `BinaryArray<i64>`",
+                        );
+                    let offsets = arrow_col.offsets().clone().into_inner();
+                    let offsets =
+                        unsafe { std::mem::transmute::<Buffer<i64>, Buffer<u64>>(offsets) };
+                    Column::Geography(GeographyColumn(BinaryColumn::new(
```

Comment:
better to use `BinaryColumn` directly without a wrapper?

Reply:
https://github.com/datafuselabs/databend/issues/14553